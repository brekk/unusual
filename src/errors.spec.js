import CONSTANTS from './constants'
import { ERRORS, throwOnInvalidInteger, testValidInteger } from './errors'

test('constants', () => {
  expect(ERRORS).toMatchSnapshot()
})

test('throwOnInvalidInteger', () => {
  expect(() => throwOnInvalidInteger(CONSTANTS.MAX_INT + 1)).toThrow()
  expect(() => throwOnInvalidInteger(CONSTANTS.MIN_INT - 1)).toThrow()
  expect(() => throwOnInvalidInteger(0)).not.toThrow()
})

test('testValidInteger', () => {
  expect(testValidInteger(CONSTANTS.MAX_INT + 1)).toEqual('TOO_BIG')
  expect(testValidInteger(CONSTANTS.MIN_INT - 1)).toEqual('TOO_SMALL')
  expect(testValidInteger(0)).toBeFalsy()
})
