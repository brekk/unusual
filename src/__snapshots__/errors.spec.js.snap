// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants 1`] = `
{
  "MIN_UNDER_MAX": "Minimum must be smaller than maximum!",
  "TOO_BIG": "Number exceeds acceptable JavaScript integer size!",
  "TOO_SMALL": "Number is below acceptable JavaScript integer size!",
}
`;
